pkg/operator/controller/gatewayclass/servicemeshcontrolplane.go

```diff
@@ -24,7 +24,7 @@ import (
 func (r *reconciler) ensureServiceMeshControlPlane(ctx context.Context, gatewayclass *gatewayapiv1beta1.GatewayClass) (bool, *maistrav2.ServiceMeshControlPlane, error) {
 	name := controller.ServiceMeshControlPlaneName(r.config.OperandNamespace)
 	have, current, err := r.currentServiceMeshControlPlane(ctx, name)
-	if err != nil {
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
I think we can drop this change and the equivalent in `subscription.go`; the existing code is consistent with other `ensureX` and `currentX` functions elsewhere in this repo.